src/ucs/datastruct/ptr_array.c

```diff
@@ -174,26 +195,43 @@ unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value,
     /* Get the first item on the free list */
     index = ptr_array->freelist;
     ucs_assert(index != UCS_PTR_ARRAY_SENTINEL);
-    elem = &ptr_array->start[index];
 
-    /* Remove from free list */
-    ptr_array->freelist = ucs_ptr_array_freelist_get_next(*elem);
+    elem = &ptr_array->start[index];
 
-    /* Populate */
-    *placeholder_p = ucs_ptr_array_placeholder_get(*elem);
+    /* Remove from free list and populate */
+    ptr_array->freelist =
```

Comment:
can use ` ucs_ptr_array_freelist_get_next` instead

Reply:
Regarding the \ in the foreach ==> This is mandatory of course, since the foreach should allow the user to write code for each element :-).

Regarding: can use ucs_ptr_array_freelist_get_next instead - Agreed and fixed.

Reply:
Regarding the \ in the foreach. Why is this mandatory? It is the last line of the macro,
As an example see `ucs_list_for_each`